spec/helpers/scm_helper_spec.rb

```diff
@@ -0,0 +1,207 @@
+
+      it('should be possible to build the source URL without the line when missing', probedock: { key: 'xdwl' }) do
+        result.custom_values['file.line'] = nil
+        expect(build_source_url(result, project, payload)).to eq('https://github.com/probedock/galactic-empire/blob/9f4b549ce75ebe92ac1bc4b697934ca1c64d7deb/path/to/the/file')
+      end
+    end
+
+    context 'gitlab' do
+      let(:project) { build(:project, repo_url: 'https://gitlab.com/probedock/galactic-empire') }
+      let(:result) { build(:result, custom_values: result_custom_values ) }
+      let(:payload) do
+        create(:test_payload, { contents_bytesize: 1, contents: { context: scm_context } })
+        TestPayload.with_scm_data.first
+      end
+
+      it('should be possible to build the source URL', probedock: { key: 'lupe' }) do
+        expect(build_source_url(result, project, payload)).to eq('https://gitlab.com/probedock/galactic-empire/blob/9f4b549ce75ebe92ac1bc4b697934ca1c64d7deb/path/to/the/file#L12')
+      end
+
+      it('should not be possible to build the source url when commit or file path is missing', probedock: { key: '6hmz' }) do
+        result.custom_values['file.path'] = nil
+        expect(build_source_url(result, project, payload)).to be_nil
+
+        result.custom_values['file.path'] = 'path'
+        payload.scm_commit = nil
+        expect(build_source_url(result, project, payload)).to be_nil
+      end
+
+      it('should be possible to build the source URL without the line when missing', probedock: { key: '6wtu' }) do
+        result.custom_values['file.line'] = nil
+        expect(build_source_url(result, project, payload)).to eq('https://gitlab.com/probedock/galactic-empire/blob/9f4b549ce75ebe92ac1bc4b697934ca1c64d7deb/path/to/the/file')
+      end
+    end
+
+    context 'bitbucket' do
+      let(:project) { build(:project, repo_url: 'https://bitbucket.com/probedock/galactic-empire') }
+      let(:result) { build(:result, custom_values: result_custom_values) }
+      let(:payload) do
+        create(:test_payload, { contents_bytesize: 1, contents: { context: scm_context } })
+        TestPayload.with_scm_data.first
+      end
+
+      it('should be possible to build the source URL', probedock: { key: '1vv1' }) do
+        expect(build_source_url(result, project, payload)).to eq('https://bitbucket.com/probedock/galactic-empire/src/9f4b549ce75ebe92ac1bc4b697934ca1c64d7deb/path/to/the/file#file-12')
+      end
+
+      it('should not be possible to build the source url when commit or file path is missing', probedock: { key: 'p8o6' }) do
+        result.custom_values['file.path'] = nil
+        expect(build_source_url(result, project, payload)).to be_nil
+
+        result.custom_values['file.path'] = 'path'
+        payload.scm_commit = nil
+        expect(build_source_url(result, project, payload)).to be_nil
+      end
+
+      it('should be possible to build the source URL without the line when missing', probedock: { key: '0tbs' }) do
```

Comment:
Please remove all the `be possible to` from test names. It's redundant and overly verbose. You in fact did it at line 145.